v2/googlecloud-to-googlecloud/src/test/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstopubsub/BigtableChangeStreamsToPubSubIT.java

```diff
@@ -0,0 +1,1021 @@
+
+    RowMutation smallMutation =
+        RowMutation.create(srcTable, rowkey)
+            .setCell(SOURCE_COLUMN_FAMILY, column, timestamp, goodValue);
+    bigtableResourceManager.write(smallMutation);
+
+    ChangelogEntryText expected =
+        ChangelogEntryMessageText.ChangelogEntryText.newBuilder()
+            .setColumn(column)
+            .setColumnFamily(SOURCE_COLUMN_FAMILY)
+            .setIsGC(false)
+            .setModType(ModType.SET_CELL)
+            .setCommitTimestamp(System.currentTimeMillis() * 1000)
+            .setRowKey(rowkey)
+            .setSourceInstance(bigtableResourceManager.getInstanceId())
+            .setSourceCluster(clusterName)
+            .setTieBreaker(1)
+            .setTimestamp(timestamp)
+            .setSourceTable(srcTable)
+            .setValue(goodValue)
+            .build();
+
+    LOG.info("Pulling 1 message from PubSub");
+    List<ReceivedMessage> receivedMessages =
+        pubsubResourceManager.getMessages(sn, 1, EXPECTED_REPLICATION_MAX_WAIT_TIME);
+    Assert.assertFalse(receivedMessages.isEmpty());
+    LOG.info("Pulled messages: {}", receivedMessages);
+
+    for (ReceivedMessage message : receivedMessages) {
+      validateJsonMessageData(expected, message.getMessage().getData().toString("UTF-8"));
+    }
+
+    Storage storage = StorageOptions.newBuilder().build().getService();
+
+    long started = System.currentTimeMillis();
+    long waitForFile = Duration.ofMinutes(30).toMillis();
+
+    String filterPrefix =
+        String.join("/", getClass().getSimpleName(), gcsClient.runId(), "dlq", "severe");
+    BlobListOption listOptions = BlobListOption.prefix(filterPrefix);
+
+    LOG.info("Looking for files with a prefix: " + filterPrefix);
+
+    boolean found = false;
+    while (!found && System.currentTimeMillis() <= (started + waitForFile)) {
+      Page<Blob> blobs = storage.list(artifactBucketName, listOptions);
+
+      for (Blob blob : blobs.iterateAll()) {
+        try {
+          byte[] content = storage.readAllBytes(blob.getBlobId());
+          ObjectMapper om = new ObjectMapper();
+          JsonNode severeError = om.readTree(content);
+          Assert.assertNotNull(severeError);
+          JsonNode errorMessageNode = severeError.get("error_message");
+          Assert.assertNotNull(errorMessageNode);
+          Assert.assertTrue(errorMessageNode instanceof TextNode);
+          String messageText = errorMessageNode.asText();
+          Assert.assertTrue(
+              "Unexpected message text: " + messageText,
+              StringUtils.contains(messageText, "Request payload size exceeds the limit"));
+          found = true;
+        } catch (Exception e) {
+          //
+          LOG.warn("Error during processing DLQ file: ", e);
+        }
+      }
+
+      if (!found) {
+        Thread.sleep(1000);
+      }
+    }
+
+    if (!found) {
+      Assert.fail("The failed message was not found in DLQ");
+    }
```

Comment:
Similar to the PubsubMessagesCheck, I would create a lambda to check for GCS files, something like:
```suggestion
    Result result =
        pipelineOperator()
            .waitForConditionAndFinish(
                createConfig(info),
                () -> {
                  List<Artifact> artifacts = gcsClient.listArtifacts(testName, expectedFilePattern);
                  for (Artifact artifact : artifacts) {
                      if (new String(artifact.contents(), StandardCharsets.UTF_8).contains("Request payload size exceeds the limit")) {
                          return true;
                      }
                  }
                  return false;
                });

    assertThatResult(result).meetsConditions();
```

Reply:
I'll take a look, thanks